pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -80,6 +86,44 @@ func (ku *Upgrader) Validate() error {
 	return nil
 }
 
+func (ku *Upgrader) setActualNodeCountForVMSSAgentPool() map[string]int {
+	nodeCountInProfile := make(map[string]int)
+	agentPoolScaleSetsToUpgradeMap := make(map[string]AgentPoolScaleSet)
+
+	for _, vmssToUpgrade := range ku.ClusterTopology.AgentPoolScaleSetsToUpgrade {
+		poolName, _, _ := utils.VmssNameParts(vmssToUpgrade.Name)
+		agentPoolScaleSetsToUpgradeMap[poolName] = vmssToUpgrade
+	}
+
+	// If the agent pool is VMSS agent pool, the actual number of node in VMSS might be different from the node count in the profile.
```

Comment:
If the vmss is under scaling, though the capacity value is right, the actual virtual machine count may be still different. Do we need to wait for scaling finish?